### PR TITLE
feat: custom short link (back-end)

### LINKED ISSUE #33: feat: custom short link (back-end)
**Task**
What is the task? Adding the possibility to write your own short link (for example with bit.lies: bit.ly/Riverz instead of bit.ly/HfDMoP).

**More about the task (optional)**
I'm only planning to add this option server-side/back-end.

**Alternatives you've considered (optional)**

**Additional context (optional)**

### PR DESCRIPTION
**Pull request by MaestroDagan**
closes #33 

## Summary
- if custom short link is provided it'll use that
- if custom short link is already used it'll respond with a 409 Conflict status code
- custom short link has basic validation
- updated docs
## Checklist
- [ ] fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] tested locally.
- [x] No error nor warning in the console.

### CODE REVIEW

File: server/src/controllers/links.controller.js [hunk 1]
```diff
@@ -1,44 +1,50 @@
...
+      .json({ success: false, message: "URL is required." });
+  }
+  if (!/(www|http:|https:)+[^\s]+[\w]/g.test(req.body.url)) {
+    return res.status(500).json({ success: false, message: "invalid url" });
+  }
+  if (await Link.findOne({ short: req.body.short }))
+    return res
+      .status(409)
+      .json({ success: false, message: "short URL already exists" });
+  const randomString = req.body.short ?? (await getRandomString());
```
Comment (cherylli) [inline server/src/controllers/links.controller.js:31]: I would rename this variable to shortCode or something, since it's not completely random anymore.

**Review by cherylli — CHANGES_REQUESTED** [review]
Could you also add 
1. basic validation of custom short links (min, max length, has to be valid characters) - right now, and empty string is accepted and saved in the database
2. edit the api documentation (swagger - docs.yaml) to reflect this change

File: server/src/controllers/links.controller.js (hunk 1 above)
Comment (MaestroDagan) [reply server/src/controllers/links.controller.js]: I've added all of the requested changes

**Comment by cherylli** [comment]
Does index only consist of formatting change? If so, could you please remove from staged
Do you also mind reverting the formatting of existing code?
Thank you

**Comment by MaestroDagan** [comment]
> 

I'll do that, thanks for your suggestions.
I think this is a really nice project to work on.

**Comment by MaestroDagan** [comment]
Removed almost all the formatting!

**Comment by MaestroDagan** [comment]
what happened?

**Comment by MaestroDagan** [comment]
Because i renamed the branche te pr closed

**Comment by MaestroDagan** [comment]
i reopened it, sorry for this.

**Comment by cherylli** [comment]
hello again, could you please check if "“Allow edits from maintainers.”  is selected in this pr. I found some errors and would like to push to your branch

**Comment by MaestroDagan** [comment]
Okay, I'll see

**Comment by MaestroDagan** [comment]
Where can i find that setting? @cherylli

**Comment by cherylli** [comment]
image

**Comment by MaestroDagan** [comment]
i dont think i can still change that

**Comment by MaestroDagan** [comment]
but i could open a new PR with that checked, do you want me to do that? @cherylli

**Comment by cherylli** [comment]
> but i could open a new PR with that checked, do you want me to do that? @cherylli

no please give me some time. I think I have messed up with the push

**Comment by MaestroDagan** [comment]
Okay!

**Comment by cherylli** [comment]
ok i think that should work
Just a few comments. It feels like you did not test the code as there were quite a few signifant errors
1. the regex didn't work, everything I tried was invalid
2. when a request was sent without "short", it gives an error because req.body.short doesn't exist, you should always check if a variable exist before using it or use optional chaining in some cases
3. some statuscode were inaccurate

When you do a pull request it's very important to check that everything is working

Let me do a final check to make sure my commit is working

**Pull request merged**